### PR TITLE
Fix longstanding compile issue on main branch

### PR DESCRIPTION
**Pull request by walterfb**
Fix longstanding compile issue on main branch by reverting two changes back to 2.8.1

### CODE REVIEW

**Review by shaoguangleo — APPROVED** [review]
It works.

File: applications/difx2fits/src/sniffer.c [hunk 1]
```diff
@@ -524,24 +502,11 @@ void deleteSniffer(Sniffer *S)
 {
 	if(S)
 	{
-		if(S->filebase)
-		{
-			free(S->filebase);
-		}
-		if(S->bp)
```
Comment (adamdeller) [inline applications/difx2fits/src/sniffer.c:531]: I think these lines related to S->bp were actually here in 2.8.1? I.e. the block about dumping bandpasses didn't need to be deleted? Maybe there were multiple commits related to this and I'm missing something. I trust Walter with handling the sniffer in any case!

File: applications/difx2fits/src/sniffer.c [hunk 2]
```diff
@@ -1183,53 +1148,9 @@ int feedSnifferFITS(Sniffer *S, const DifxVis *dv)
...
-		}
-		else
-		{
-			writeNewConfig = 0;
-		}
-
 		S->nIF = dfs->nIF;
 		S->nPol = dc->nPol;
 		S->configId = configId;
-		++S->nConfigsUsed;
```
Comment (adamdeller) [inline applications/difx2fits/src/sniffer.c:1202]: Similarly here I thought most of this code actually was present, and it was only from line 1220 that was actually the edited/added code. I'm looking at commit 11039 from SVN (Walter, 15 Aug 2023).

**Review by adamdeller — COMMENTED** [review]
I think it's fine but maybe Walter can double check whether those extra deletions from the sniffer are intended?

**Pull request merged**